### DISCUSSION TITLE
POST to mongoDB doesnt transfer data

### DISCUSSION THREAD

**Discussion by t0bbl** [neuefische/questions#590] [Web] [answered]
## 🎉 Introduction:

Iam sending a POST to my mongoDB but it doesnt past the data i want it to. 

## 🧮 Pull Request or link to code:

[PR](https://github.com/t0bbl/capstone-project/pull/14)

![Screenshot 2023-05-03 at 10 08 52](https://user-images.githubusercontent.com/117731660/235862995-4fc6e24d-461a-41a1-b145-03e1372ca452.png)

## ♻️ What are the steps to reproduce this issue?

1. put three Keywords in the Form to send a POST
2. get a console.log "shirt created from sendRequest"
3. profit (or not ...)

## 🤔 What happens?

The form.js uses a SWR trigger to start the sendRequest and "ping" the api which is important for a later part of the project. A new entry in the MongoDB is made but the shirtdata isnt past correctly. The new DB entry misses the keywords one two and three which should be given through the sendRequest which contains a json.stringifiy for the shirtData. You are send to /ChooseFour/${id} thanks to the router.push in the handleSubmit.

## 👉 What did you expect to happen?

I expected the new DB entry to contain the keywords keywordOne, keywordTwo, keywordThree.

## 🤖 Any logs, error output, etc?

The console is giving out errors for the pictures missing on the follow up page, which is ok for now. There is a log from the sendRequest when its run through.

## ✍️ Any other comments?

Thanks for having a look.

## 🔎 Search engine keywords, documentation, resources

I looked through the mongoDB docs, which werent really helpful. I checked the corresponding Handouts but cant find the help iam looking for.

**Comment by NikRoe** [comment, marked as answer]
Hi Robin,

thanks for moving the question :) 

Your code looks 99% correct! However, there is a small issue in your `sendRequest()`, it is required for useSWRMutation to call the second argument { arg }.

If you change that function to this, it should work:
```js
async function sendRequest(url, { arg }) {
  const response = await fetch(url, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(arg),
  });
  const { status } = await response.json();
  console.log(status + " from sendRequest");
}
```

Hope this helps :)

**Reply by t0bbl** [reply]
there was also a small bug in my .env. after your fix a error message was displayed in the server sided console which helped. 
thanks!
